apps/ui/components/reservation/EditStep0.tsx

```diff
@@ -121,41 +134,149 @@ const EditStep0 = ({
   activeApplicationRounds,
   setErrorMsg,
   nextStep,
-  apiBaseUrl,
   isLoading,
 }: Props): JSX.Element => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
-
   const isMobile = useMedia(`(max-width: ${breakpoints.m})`, false);
-
-  const [focusDate, setFocusDate] = useState(new Date(reservation.begin));
   const [calendarViewType, setCalendarViewType] = useState<WeekOptions>("week");
 
   const isClientATouchDevice = isTouchDevice();
   const [shouldCalendarControlsBeVisible, setShouldCalendarControlsBeVisible] =
     useState(false);
+  const startDate = useMemo(
+    () => new Date(reservation.begin),
+    [reservation.begin]
+  );
+  const reservationForm = useForm<PendingReservationFormType>({
+    defaultValues: {
+      date: reservation.begin,
```

Comment:
Date format? here it's ISO string (directly from backend). 
Should be in UI format otherwise it's going to break HDS DateInput.